app/controllers/friendships_controller.rb

```diff
@@ -43,12 +40,6 @@ def destroy
 
```

Comment:
To make your code clearer and more effective, move model manipulation into model class. 

Your inverse friendship method should be created in the Friendship model. and might be similar to the snippet below:
```ruby
# Friendship model

def confirm_friend
  self.update_attributes(confirmed: true)
  Friendship.create!(friend_id: self.user_id,
                  user_id: self.friend_id,
                  confirmed: true)
end

```

You can then call the `confirm_friend` method in the update action of your friendship controller


if your `confirm_friend` method is similar to the one above, you should be able to find user friends only by user_id in the friendships table:

```ruby
# User model

has_many :confirmed_friendships, -> { where confirmed: true }, class_name: "Friendship"
has_many :friends, through: :confirmed_friendships
```

Reply:
Fixed